sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1356,6 +1366,11 @@ static void *_gnrc_netif_thread(void *args)
                     DEBUG("gnrc_netif: error sending packet %p (code: %u)\n",
                           msg.content.ptr, res);
                 }
+#ifdef MODULE_NETSTATS_L2
+                else {
+                    netif->stats.tx_bytes += res;
```

Comment:
Was this subtle difference in behavior intentional?
I understand that it is not desirable to "waste" bytes in the netif struct to just pass the frame size from here to the TX feedback event handler, but this does change the current behavior.
number of bytes sent are increased when preloading the frame before sending, while frame count is increased upon TX_COMPLETE feedback event. This means that when using NETOPT_PRELOADING, we will get the byte count of all preloaded frames, while the number of transmissions are the number of frames actually sent. This can be both more or less than the number of preloaded frames since it is possible to transmit the same preloaded frame multiple times, as well as preloading another frame without sending the first at all.

Reply:
mhm, a change in behaviour was not intended as such. And looking for instance at the cc2538rf implementation there is no difference, at lease not to me. In the `_send()` function of that driver it was

```
netdev->stats.tx_bytes += pkt_len;
[...]
return pkt_len;
```

and here 

```
res = netif->ops->send()
[...]
netdev->stats.tx_bytes += res;
```

so for this driver it should be the same, right? I'll check the others, too - where do you see the diff exactly?

Reply:
Sorry for the noise, I had somehow misread the code and I thought that the rx_bytes increment used to be in the tx_exec function..

There is a time delay between the increment of stats.tx_bytes until the increment of stats.tx_complete, but it is the same in the original code as in this PR.